crates/bevy_ecs/macros/src/fetch.rs

```diff
@@ -0,0 +1,740 @@
+use proc_macro::TokenStream;
+use proc_macro2::{Ident, Span};
+use quote::{quote, ToTokens};
+use syn::{
+    parse::{Parse, ParseStream},
+    parse_macro_input, parse_quote,
+    punctuated::Punctuated,
+    Attribute, Data, DataStruct, DeriveInput, Field, Fields, GenericArgument, GenericParam,
+    ImplGenerics, Lifetime, LifetimeDef, Path, PathArguments, ReturnType, Token, Type,
+    TypeGenerics, TypePath, Visibility, WhereClause,
+};
+
+use crate::bevy_ecs_path;
+
+#[derive(Default)]
+struct FetchStructAttributes {
+    pub mutable: bool,
+    pub read_only_derive_args: Punctuated<syn::NestedMeta, syn::token::Comma>,
+}
+
+static MUTABLE_ATTRIBUTE_NAME: &str = "mutable";
+static READ_ONLY_DERIVE_ATTRIBUTE_NAME: &str = "read_only_derive";
+
+mod field_attr_keywords {
+    syn::custom_keyword!(ignore);
+}
+
+static FETCH_ATTRIBUTE_NAME: &str = "fetch";
+static FILTER_FETCH_ATTRIBUTE_NAME: &str = "filter_fetch";
+
+pub fn derive_fetch_impl(input: TokenStream) -> TokenStream {
+    let ast = parse_macro_input!(input as DeriveInput);
+
+    let mut fetch_struct_attributes = FetchStructAttributes::default();
+    for attr in &ast.attrs {
+        if !attr
+            .path
+            .get_ident()
+            .map_or(false, |ident| ident == FETCH_ATTRIBUTE_NAME)
+        {
+            continue;
+        }
+
+        attr.parse_args_with(|input: ParseStream| {
+            let meta = input.parse_terminated::<syn::Meta, syn::token::Comma>(syn::Meta::parse)?;
+            for meta in meta {
+                let ident = meta.path().get_ident();
+                if ident.map_or(false, |ident| ident == MUTABLE_ATTRIBUTE_NAME) {
+                    if let syn::Meta::Path(_) = meta {
+                        fetch_struct_attributes.mutable = true;
+                    } else {
+                        panic!(
+                            "The `{}` attribute is expected to have no value or arguments",
+                            READ_ONLY_DERIVE_ATTRIBUTE_NAME
+                        );
```

Comment:
```suggestion
                if ident.map_or(false, |ident| ident == MUTABLE_ATTRIBUTE_NAME) {
                    if let syn::Meta::Path(_) = meta {
                        fetch_struct_attributes.mutable = true;
                    } else {
                        panic!(
                            "The `{}` attribute is expected to have no value or arguments",
                            MUTABLE_ATTRIBUTE_NAME
                        );
```

Reply:
Yeah, noticed it as well and force-pushed the fix. Thank you :)